bootstrap/interfaces/secret.go

```diff
@@ -28,4 +28,10 @@ type SecretProvider interface {
 
 	// HasSecret returns true if the service's SecretStore contains a secret at the specified path.
 	HasSecret(path string) (bool, error)
+
+	// RegisteredSecretUpdateCallback registers a callback for a secret.
+	RegisteredSecretUpdateCallback(path string, callback func(path string))
+
+	// SecretsUpdatedWithPath .
+	SecretsUpdatedWithPath(path string)
```

Comment:
```suggestion
	SecretUpdatedAtPath(path string)
```

However, should this really be a public interface method?  Isn't this internal detail?  Why would a user of the interface need to call this method?

Reply:
Per my other comments `SecretUpdatedAtPath` will need to be called from the config processing outside this package so must be public. And it replaces `SecretsUpdated` which we may be able to remove.